integration-tests/test_connection.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import conftest
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_connection(insights_client):
+    """
+    Test insights-client --test-connection
+    `test_connection` method returns 0 if success, 1 if failure
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_http_timeout(insights_client):
+    """
+    Test http_timeout config option
+    Set http_timeout to a very low value, run --test-connection and check
+    the set time
+    """
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    insights_client.config.http_timeout = 0.1
+    insights_client.config.save()
+
+    output = insights_client.run("--test-connection", check=False)
+    assert output.returncode == 1
+    assert "read timeout=0.1" in output.stdout
+
+
+def test_noauth_proxy_connection(insights_client, test_config):
+    """
+    Test no-auth proxy option with --test-connection option
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
```

Comment:
similar to what Archana already pointed out, there is no need to register for this test to run